com.unity.ml-agents.extensions/Runtime/Sensors/GridSensorComponent.cs

```diff
@@ -202,68 +194,129 @@ public int ObservationStacks
             set { m_ObservationStacks = value; }
         }
 
+        [HideInInspector, SerializeField]
+        internal bool m_UseOneHotTag = true;
+        /// <summary>
+        /// Whether to use one-hot detected tag as observation.
+        /// Note that changing this after the sensor is created has no effect.
+        /// </summary>
+        public bool UseOneHotTag
+        {
+            get { return m_UseOneHotTag; }
+            set { m_UseOneHotTag = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        internal bool m_CountColliders = false;
+        /// <summary>
+        /// Whether to use the number of count for each detectable tag as observation.
+        /// Note that changing this after the sensor is created has no effect.
+        /// </summary>
+        public bool CountColliders
+        {
+            get { return m_CountColliders; }
+            set { m_CountColliders = value; }
+        }
+
         /// <inheritdoc/>
         public override ISensor[] CreateSensors()
         {
-            m_Sensor = new GridSensor(
-                m_SensorName,
+            List<ISensor> m_Sensors = new List<ISensor>();
+            m_BoxOverlapChecker = new BoxOverlapChecker(
                 m_CellScale,
                 m_GridSize,
                 m_RotateWithAgent,
-                m_ChannelDepths,
-                m_DetectableObjects,
                 m_ColliderMask,
-                m_DepthType,
                 gameObject,
-                m_CompressionType,
-                m_MaxColliderBufferSize,
-                m_InitialColliderBufferSize
+                m_DetectableObjects,
+                m_InitialColliderBufferSize,
+                m_MaxColliderBufferSize
             );
 
-            if (ObservationStacks != 1)
+            if (m_UseOneHotTag)
             {
-                return new ISensor[] { new StackingSensor(m_Sensor, ObservationStacks) };
+                var sensor = new OneHotGridSensor(m_SensorName, m_CellScale, m_GridSize, m_ChannelDepths, m_DetectableObjects, m_CompressionType);
```

Comment:
note that you'll need to append something to the name either here or in the implementations, otherwise you can get two sensors with the same name.

Reply:
added